File: 1787-sum-of-absolute-differences-in-a-sorted-array/sum-of-absolute-differences-in-a-sorted-array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        n = len(nums)
        total_sum = sum(nums)
        left_sum = 0
        result = []
        
        for i, val in enumerate(nums):
            right_sum = total_sum - left_sum - val
            
            left_total = (i * val) - left_sum
            right_total = right_sum - ((n - 1 - i) * val)
            
            result.append(left_total + right_total)
            
            left_sum += val
            
        return result

File: 1787-sum-of-absolute-differences-in-a-sorted-array/sum-of-absolute-differences-in-a-sorted-array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        vector<int> res;
        int prefixSum = 0;
        int suffixSum = 0;

        for (int num : nums) {
            suffixSum += num;
        }

        for (int i = 0; i < nums.size(); i++) {
            int leftSum = nums[i] * i - prefixSum;
            int rightSum = suffixSum - nums[i] * (nums.size() - i);
            int totalSum = leftSum + rightSum;

            res.push_back(totalSum);

            prefixSum += nums[i];
            suffixSum -= nums[i];
        }

        return res;        
    }
};